### PR TITLE
Limited shm frame count

### PR DESCRIPTION
**Pull request by NickM-27**
This is a simpler approach to https://github.com/blakeblackshear/frigate/pull/12020 which has the capture process clean up SHM frames automatically to ensure the SHM will not overflow.

### CODE REVIEW

File: frigate/object_processing.py [hunk 1]
```diff
@@ -659,9 +659,14 @@ def on(self, event_type: str, callback: Callable[[dict], None]):
...
-            frame_id, self.camera_config.frame_shape_yuv
-        )
+
+        try:
+            current_frame = self.frame_manager.get(
+                frame_id, self.camera_config.frame_shape_yuv
+            )
+        except FileNotFoundError:
+            logger.error(f"Failed to get {frame_id} from SHM")
+            return
```
Comment (blakeblackshear) [inline frigate/object_processing.py:669]: Is return the behavior we would want here? Most of the code below doesn't depend on the frame.

File: frigate/object_processing.py [hunk 2]
```diff
@@ -659,9 +659,14 @@ def on(self, event_type: str, callback: Callable[[dict], None]):
     def update(self, frame_time, current_detections, motion_boxes, regions):
         # get the new frame
         frame_id = f"{self.name}{frame_time}"
-        current_frame = self.frame_manager.get(
-            frame_id, self.camera_config.frame_shape_yuv
-        )
+
+        try:
```
Comment (blakeblackshear) [inline frigate/object_processing.py:663]: Should we put this try/except in the frame manager itself and return `None` if the frame isn't found?

**Review by blakeblackshear — APPROVED** [review]

**Comment by YBonline** [comment]
This causes my SHM to run out immediately on startup, no errors in the logs except no space available, reverting to previous build fixes issue

**Comment by NickM-27** [comment]
how much SHM do you have available and what are the detect fps?

**Comment by YBonline** [comment]
shm_size: "1521mb" 
5 fps

Honestly, thinking about it I might need to recalculate based off the guide of what it should be, I might have added cameras since then, but I've had 0 zero issues before this.
I seem to recall in the past if the shm size was too low, there would be a message in log with a recommended value, I don't seem to see that, is my memory right?

**Comment by NickM-27** [comment]
there is but technically we did not update that calculation for this, not 100% sure if things changed. I am also seeing issues but not like you are describing, it is just having a lot of missed frames

**Pull request merged**